samples/algorithms/MajoranaQubits/src/GateSet.qs

```diff
@@ -0,0 +1,62 @@
+/// A set of gates built upon the custom measurements
+/// provided by the hardware provider.
+///
+/// Source:
+///  [1] Surface code compilation via edge-disjoint paths
+///      https://arxiv.org/pdf/2110.11493
+
+/// Apply a CNOT gate to the given qubits.
+/// Source: [1] Figure 3.
+operation CNOT(control : Qubit, target : Qubit) : Unit {
+    use ancilla = Qubit();
+    let a = Mzz(control, target);
+    let b = Mxx(target, ancilla);
+    let c = Mx(target);
+    if b == One {
+        Z(control);
+    }
+    if [a, c] == [One, One] {
+        X(ancilla);
+    }
+}
+
+/// Prepare a Bell Pair.
+/// Source: [1] Figure 18a.
+operation BellPair(q1 : Qubit, q2 : Qubit) : Unit {
+    // Bring the qubits to their ground state.
```

Comment:
But you don't flip them if the Mz measures 'One', so how is this ensuring they are in ground state? (The simulatable instrinsic also doesn't ensure they are Zero)

Reply:
'Ground state' was the wrong thing to say here. Is more like it's collapsing the Z axis, projecting them to the X basis, where the next Mxx is performed.

Reply:
I should definitely understand how this works and include comments explaining it.